sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -839,6 +839,24 @@ object SQLConf {
     .checkValues(HiveCaseSensitiveInferenceMode.values.map(_.toString))
     .createWithDefault(HiveCaseSensitiveInferenceMode.NEVER_INFER.toString)
 
+  val HIVE_BLOBSTORE_SUPPORTED_SCHEMES =
+    buildConf("spark.sql.hive.blobstore.supported.schemes")
+      .doc("Comma-separated list of supported blobstore schemes.")
+      .version("3.1.0")
+      .stringConf
+      .transform(_.toLowerCase(Locale.ROOT))
+      .createWithDefault("s3,s3a,s3n")
+
+  val HIVE_BLOBSTORE_USE_BLOBSTORE_AS_SCRATCHDIR =
+    buildConf("spark.sql.hive.blobstore.use.blobstore.as.scratchdir")
```

Comment:
I'm still not sure we need to strictly follow the Hive option names in this PR. I googled these option names though, only a few valid pages found. So, how about simplifying it like this?
```
// SQLConf
  val HIVE_SUPPORTED_SCHEMES_TO_USE_NONBLOBSTORE =
    buildConf("spark.sql.hive.supportedSchemesToUseNonBlobStore") <-- new name assigned for Spark
      .doc("<detailed description>")
      .version("3.1.0")
      .stringConf
      .transform(_.toLowerCase(Locale.ROOT))
      .createWithDefault("") <-- disabled by default

  def supportedSchemesToUseNonBlobStore : String = getConf(HIVE_SUPPORTED_SCHEMES_TO_USE_NONBLOBSTORE)

// SaveAsHiveFile
  private def supportSchemeToUseNonBlobStore(path: Path): Boolean = {
    path != null && {
      val supportedBlobSchemes = SQLConf.get.supportedSchemesToUseNonBlobStore
      val scheme = Option(path.toUri.getScheme).getOrElse("")
      Utils.stringToSeq(supportedBlobSchemes).contains(scheme.toLowerCase(Locale.ROOT))
    }
  }

      if (supportSchemeToUseNonBlobStore(path)) {
        // Hive sets session_path as HDFS_SESSION_PATH_KEY(_hive.hdfs.session.path) in hive config
        val sessionScratchDir = externalCatalog.unwrapped.asInstanceOf[HiveExternalCatalog]
          .client.getConf("_hive.hdfs.session.path", "")
        logDebug(s"session scratch dir '$sessionScratchDir' is used")
        getMRTmpPath(hadoopConf, sessionScratchDir, scratchDir)
      } else {
        newVersionExternalTempPath(path, hadoopConf, stagingDir)
      }
```

Reply:
I modified like below
- `spark.sql.hive.blobstore.supported.schemes` -> `spark.sql.hive.supportedSchemesToUseNonBlobStore` (as you suggested)
- `spark.sql.hive.blobstore.use.blobstore.as.scratchdir` -> `spark.sql.hive.useBlobstoreAsScratchDir`

Reply:
I merged the two options into `spark.sql.hive.supportedSchemesToUseNonBlobStore`